micro_dl/inference/inference.py

```diff
@@ -316,364 +169,181 @@ def predict_large_image(self, input_image, model=None):
-        prediction,
-        metrics_list,
-        metrics_orientations,
-        path="unspecified",
-        window=None,
-    ):
-        """
-        Gets metrics for this target_/prediction pair in all the specified orientations for all the
-        specified metrics.
-
-        :param np.ndarray target: 5d target array (on cpu)
-        :param np.ndarray prediction: 5d prediction array (on cpu)
-        :param list metrics_list: list of strings indicating the name of a desired metric, for options
-                                    see inference.evaluation_metrics. MetricsEstimator docstring
-        :param list metrics_orientations: list of strings indicating the orientation to compute, for
-                                    options see inference.evaluation_metrics. MetricsEstimator docstring
-        :param tuple window: spatial window of this target/prediction pair in the larger arrays they
-                                    come from.
-
-        :return dict prediction_metrics: dict mapping orientation -> pd.dataframe of metrics for that
-                                    orientation
-        """
-        metrics_estimator = inference_metrics.MetricsEstimator(metrics_list)
-        prediction_metrics = {}
-
-        # transpose target and prediction to be in xyz format
-        # NOTE: This expects target and pred to be in the format bczyx!
-        target = np.transpose(target, (0, 1, -2, -1, -3))
-        prediction = np.transpose(prediction, (0, 1, -2, -1, -3))
-
-        zstart, zend = window[0][0], window[0][0] + window[1][0]  # end = start + length
-        pred_name = f"slice_{zstart}-{zend}"
-
-        if "xy" in metrics_orientations:
-            metrics_estimator.estimate_xy_metrics(
-                target=target,
-                prediction=prediction,
-                pred_name=pred_name,
-            )
-            metrics_xy = self._collapse_metrics_dict(
-                metrics_estimator.get_metrics_xy().to_dict()
+        # Positions are specified
+        if isinstance(self.inference_config["positions"], dict):
+            print("Predicting on positions specified in inference config.")
+            return self.inference_config["positions"]
+        elif not isinstance(self.inference_config["positions"], str):
+            raise AttributeError(
+                "Invalid 'positions'. Expected one of {str, dict}"
+                f" but got {self.inference_config['positions']}"
             )
-            prediction_metrics["xy"] = metrics_xy
 
-        if "xyz" in metrics_orientations:
-            metrics_estimator.estimate_xyz_metrics(
-                target=target,
-                prediction=prediction,
-                pred_name=pred_name,
-            )
-            metrics_xyz = self._collapse_metrics_dict(
-                metrics_estimator.get_metrics_xyz().to_dict()
-            )
-            prediction_metrics["xyz"] = metrics_xyz
+        #Positions are unspecified and need to be read
```

Comment:
Currently the inference reads the positions specified in the inference section of the config yaml file and then proceeds to the following lines, where it also reads the data split yaml stored in the model directory specified. Should it be modified to chose either, based on decision 
- if we run it on a new dataset, take the position list mentioned in the config.
- if we run it on test set from the dataset used for generating the model, then use the data split file?